src/lib/foreign-field.ts

```diff
@@ -497,6 +494,18 @@ class AlmostForeignField extends ForeignFieldWithMul {
   static unsafeFrom(x: ForeignField) {
     return new this(x.value);
   }
+
+  /**
+   * Check equality with a constant value.
+   *
+   * @example
+   * ```ts
+   * let isXZero = x.equals(0);
+   * ```
+   */
+  equals(y: bigint | number) {
```

Comment:
I noticed (when I wanted to use it) that the original `ForeignField.equals()` is very error-prone, since it checked for exact equality instead of equality mod p. I replaced it with this constant version (which assumes an almost reduced input) and the full version on `CanonicalForeignField`